src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler.properties

```
@@ -3228,15 +3228,18 @@ compiler.err.this.as.identifier=\
 compiler.err.receiver.parameter.not.applicable.constructor.toplevel.class=\
     receiver parameter not applicable for constructor of top-level class
 
-# TODO 308: make a better error message
+# 0: fragment, 1: symbol, 2: type
+compiler.err.type.annotation.inadmissible=\
+    type annotations are not admissible at this location: {0}\n\
```

```suggestion
compiler.err.type.annotation.not.applicable=\
    type annotations are not applicable at this location: {0}\n\
```

I don't know whether there are concrete guidelines for this or whether "admissible" is what the JLS uses for this.
Looking at other annotation-related error messages in this file, most use "applicable" and "not applicable" (one key uses `inapplicable`, but the message still uses `not applicable`).
There are no other uses of `admissible` in this file.
There is also `annotation.not.valid.for.type` which uses `annotation not valid for an element of type {0}`, which could be another option.

Also, not not sure whether it would be `are not applicable *at* this location`, `are not applicable *on* this location`, or something else.

---

Thanks, I'm very open to changing the language here. I was using 'admissible' because it is used in [JVMS 4.7.20](https://docs.oracle.com/javase/specs/jvms/se21/html/jvms-4.html#jvms-4.7.20) and [JLS 9.7.4](https://docs.oracle.com/javase/specs/jls/se21/html/jls-9.html#jls-9.7.4) to describe where type annotations are allowed on nested types.